recipe/quadrature/quadrature.cc

```diff
@@ -0,0 +1,39 @@
+#include "recipe/quadrature/quadrature.h"
+#include <cassert>
+
+namespace recipe {
+namespace quadrature {
```

Comment:
Maybe we can use `integrate` for a more generic namespace. There are a couple of algorithms which is not categorized as `quadrature` such as `monte-carlo` and `quasi monte-carlo`. Then `quadrature` can be sub-namespace of it.